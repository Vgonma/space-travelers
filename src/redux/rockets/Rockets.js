import axios from 'axios';

const GETDATA = 'react_redux_project/rockets/GETDATA';
const ROCKETS_URL = 'https://api.spacexdata.com/v3/rockets';

export const getRockets = () => async (dispatch) => {
  const response = await axios.get(ROCKETS_URL)
    .then((allRockets) => Object.entries(allRockets.data).map(([id, rockets]) => (
      { id, ...rockets[0] })));
  dispatch({
    type: GETDATA,
    payload: response,
  });
};

const rocketReducer = (state = [], action) => {
  switch (action.type) {
    case GETDATA: return action.payload;
    default: return state;
  }
};

export default rocketReducer;
