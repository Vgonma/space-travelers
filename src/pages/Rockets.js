import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import RocketList from '../components/rocket/RocketList';
import { getRockets } from '../redux/rockets/Rockets';
import '../styles/rockets.scss';

let flag = false;
const Rockets = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    if (!flag) {
      dispatch(getRockets());
      flag = true;
    }
  }, [dispatch]);
  return (
    <section className="rocket-container container">
      <ul>
        <RocketList />
      </ul>
    </section>
  );
};

export default Rockets;
