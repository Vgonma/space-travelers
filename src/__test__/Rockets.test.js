import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Rockets from '../pages/Rockets';

describe('Rockets', () => {
  test('renders Rockets Page without crashing', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Rockets />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree.toJSON).toMatchSnapshot();
  });

  test('renders Rockets Page without crashing', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Rockets />
        </BrowserRouter>
      </Provider>,
    )
    container.querySelector('.rocket-name')
    expect(container.firstChild).toMatchSnapshot()

    container.querySelector('.rocket-detail')
    expect(container.firstChild).toMatchSnapshot()
  });
});
