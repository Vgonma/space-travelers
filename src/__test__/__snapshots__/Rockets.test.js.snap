// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets renders Rockets Page without crashing 1`] = `undefined`;

exports[`Rockets renders Rockets Page without crashing 2`] = `
<section
  class="rocket-container container"
>
  <ul>
    <div
      class="rocket_container"
    />
  </ul>
</section>
`;

exports[`Rockets renders Rockets Page without crashing 3`] = `
<section
  class="rocket-container container"
>
  <ul>
    <div
      class="rocket_container"
    />
  </ul>
</section>
`;
